### PR TITLE
Add option to bug_classifier script to download model when it doesn't exist

### LINKED ISSUE #832: Add an option to the bug_classifier script to download the model when it doesn't exist

### PR DESCRIPTION
**Pull request by chidauri**
Fixes #832

### CODE REVIEW

File: scripts/bug_classifier.py [hunk 1]
```diff
@@ -22,9 +24,23 @@ def classify_bugs(model_name, classifier):
     else:
         model_file_name = f"{model_name}model"
 
-    assert os.path.exists(
-        model_file_name
-    ), f"{model_file_name} does not exist. Train the model with trainer.py first."
+    if not os.path.exists(model_file_name):
```
Comment (marco-c) [inline scripts/bug_classifier.py]: You should use the functions in `utils` for downloading and for extracting.

File: scripts/bug_classifier.py [hunk 2]
```diff
@@ -22,9 +23,12 @@ def classify_bugs(model_name, classifier):
...
+        print(f"{model_file_name} does not exist. Downloading the model....")
```
Comment (marco-c) [inline scripts/bug_classifier.py]: Nit: use logger.info (you'll also need to setup the logger like other scripts do).

File: scripts/bug_classifier.py [hunk 3]
```diff
@@ -22,9 +23,12 @@ def classify_bugs(model_name, classifier):
...
+        download_url = f"https://index.taskcluster.net/v1/task/project.relman.bugbug.train_{model_name}.latest/artifacts/public/{model_file_name}.zst"
```
Comment (marco-c) [inline scripts/bug_classifier.py]: Nit: I'd remove this variable and replace its usage with its value
`download_check_etag(f"https://index.tas...", f"{model_file...")`

**Comment by marco-c** [comment]
What happens after your changes if the model is not available for download?

**Comment by chidauri** [comment]
> What happens after your changes if the model is not available for download?

requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://index.taskcluster.net/v1/task/project.relman.bugbug.train_uplift.latest/artifacts/public/upliftmodel.zst

**Comment by marco-c** [comment]
> requests.exceptions.HTTPError: 404 Client Error

We should catch the exception and log the error: "A pre-trained model is not available, you will need to train it yourself using the trainer script"

File: scripts/bug_classifier.py [hunk 4]
```diff
@@ -22,9 +28,21 @@ def classify_bugs(model_name, classifier):
...
+        try:
+            download_check_etag(
+                f"https://index.taskcluster.net/v1/task/project.relman.bugbug.train_{model_name}.latest/artifacts/public/{model_file_name}.zst",
+                f"{model_file_name}.zst",
+            )
+        except requests.HTTPError:
+            logger.error(
+                f"A pre-trained model is not available, you will need to train it yourself using the trainer script"
+            )
+            raise SystemExit(0)
```
Comment (marco-c) [inline scripts/bug_classifier.py]: Since it's an error, the error code should not be 0 (0 means no error).

Comment (chidauri) [reply scripts/bug_classifier.py]: ohhh :P

**Pull request merged**